src/components/QuestDetailsWrapper/index.tsx

```diff
@@ -587,14 +592,23 @@ export function QuestDetailsWrapper({
           eligible: false,
           steamAccountLinked: true
         },
-        playStreak: getPlaystreakArgsFromQuestData(
+        playStreak: getPlaystreakArgsFromQuestData({
           questMeta,
           questPlayStreakData,
-          isSignedIn
-        )
+          useModuleInitTimeForSessionStartTime: isSignedIn,
+          onSync: async () => {
+            if (questsWithExternalPlayStreakSync.includes(questMeta.id)) {
+              await syncPlayStreakWithExternalSource(questMeta.id)
+            } else {
+              await syncPlaySession(projectId, 'hyperplay')
```

Comment:
you need to also call `resetSessionStartedTime` if you're going to call `syncPlaySession` here

Reply:
and you may need a delay between the `syncPlaySession` call and the invalidate query like in the hook:

```
      await syncPlaySession(projectId, 'hyperplay')
      // allow for some time before read
      await wait(5000)
      await invalidateQuery()
      resetSessionStartedTime()
```

Reply:
done, I'll try not to add the wait to check if we truly need it. I tested without it, and it's working fine